src/ArmTemplates/Extractor/EntityExtractors/ProductApisExtractor.cs

```diff
@@ -73,7 +77,11 @@ async Task<List<TemplateResource>> GenerateSingleApiTemplateAsync(string singleA
                 }
 
                 this.logger.LogInformation("{0} Product API found ...", singleApiName);
-                return await this.GenerateProductApiTemplateResourcesAsync(singleApiName, extractorParameters, new string[] { });
+                var dependsOnParameter = addDependsOnParameter
+                    ? new string[] { $"[resourceId('Microsoft.ApiManagement/service/apis', parameters('{ParameterNames.ApimServiceName}'), '{singleApiName}')]" }
+                    : new string[] { };
```

Comment:
```suggestion
                    : Array.Empty<string>();
```

Reply:
didn't really even pay attention to this while refactoring. Found all Extractor `new string[] { }` and changed it. Thanks!